Lib/glyphsLib/builder/user_data.py

```diff
@@ -167,8 +167,13 @@ def to_glyphs_layer_user_data(self, ufo_glyph, layer):
 
 def to_glyphs_node_user_data(self, ufo_glyph, node, path_index, node_index):
     key = f"{NODE_USER_DATA_KEY}.{path_index}.{node_index}"
+    # Remember node name because we overwrite userData, which internally stores the
+    # node name. We take the name directly from the UFO point in `to_glyphs_paths`.
+    old_name = node.name
     if key in ufo_glyph.lib:
         node.userData = ufo_glyph.lib[key]
+    if node.name is None:
+        node.name = old_name
```

Comment:
i have no idea what's going on here. what does 'overwriting node.userData' have to do with node.name attribute?

Reply:
We do `node.userData = ufo_glyph.lib[key]`. `node.name` is sugar for `node.userData.get("name", None)`. Assigning something new to userData may change the node name then. Since we take the node name from the UFO's Point.name, we must not hold on to the node name before userData assignment.

Reply:
> node.name is sugar for node.userData.get("name", None)

that's what doesn't make sense to me. why isn't it a regular attribute or read-only property stored in the node object itself?

Reply:
Glyphs stores the name of a node in userData:

```
[...]
"123 456 OFFCURVE {name = \"aaa\";}"
[...]
```
Probably was the most straightforward thing to do.